src/charm.py

```diff
@@ -544,6 +559,25 @@ def _set_n2_information(self) -> None:
             amf_port=NGAPP_PORT,
         )
 
+    def _get_telemetry_config(self) -> TelemetryConfig:
+        enabled: bool = False
+        ratio = 0.0
+        otlp_endpoint = ""
+
+        is_ready = self.tracing.is_ready()
+        endpoints = self.tracing.get_all_endpoints()
+
+        if is_ready and endpoints:
+            enabled = True
+            ratio = 1.0
+            otlp_endpoint = endpoints.receivers[0].url
```

Comment:
I think there's supposed to be a helper for that in the tracing charm lib.
The `receivers` shape is a bit odd, the logic should be like "first receiver URL among those that have the correct protocol".

Arguably the code should work as is, because only one protocol was ever requested and tracing says it's ready. At least I think it should.